lib/writer/drift_model_writer.dart

```diff
@@ -0,0 +1,255 @@
+import 'package:model_generator/config/pubspec_config.dart';
+import 'package:model_generator/config/yml_generator_config.dart';
+import 'package:model_generator/model/field.dart';
+import 'package:model_generator/model/item_type/array_type.dart';
+import 'package:model_generator/model/item_type/integer_type.dart';
+import 'package:model_generator/model/item_type/map_type.dart';
+import 'package:model_generator/model/model/object_model.dart';
+import 'package:model_generator/util/case_util.dart';
+import 'package:model_generator/util/generic_type.dart';
+import 'package:model_generator/util/type_checker.dart';
+import 'package:path/path.dart';
+
+class DriftModelWriter {
+  final PubspecConfig pubspecConfig;
+  final ObjectModel jsonModel;
+  final List<Field> extendsFields;
+  final YmlGeneratorConfig yamlConfig;
+
+  const DriftModelWriter(
+    this.pubspecConfig,
+    this.jsonModel,
+    this.extendsFields,
+    this.yamlConfig,
+  );
+
+  String write() {
+    final sb = StringBuffer();
+    final modelDirectory = [
+      pubspecConfig.projectName,
+      jsonModel.baseDirectory,
+      jsonModel.path,
+      '${jsonModel.fileName}.dart'
+    ].whereType<String>();
+    final imports = <String>{}
+      ..add("import 'package:drift/drift.dart';")
+      ..add(
+          "import 'package:${pubspecConfig.projectName}/${pubspecConfig.databasePath}';")
+      ..add("import 'package:${joinAll(modelDirectory)}';");
+
+    for (final element in (jsonModel.extraImports ?? [])) {
+      imports.add('import \'$element\';');
+    }
+    final extendsModel = jsonModel.extendsModel;
+
+    if (extendsModel != null) {
+      if (!TypeChecker.isKnownDartType(extendsModel)) {
+        imports.addAll(_getImportsFromPath(extendsModel));
+      }
+    }
+
+    for (final field in jsonModel.fields) {
+      final type = field.type;
+      if (!TypeChecker.isKnownDartType(type.name) &&
+          type.name != jsonModel.name) {
+        imports.addAll(_getImportsFromPath(type.name));
+      }
+      if (type is MapType && !TypeChecker.isKnownDartType(type.valueName)) {
+        imports.addAll(_getImportsFromPath(type.valueName));
+      }
+    }
+    for (final field in extendsFields) {
```

Comment:
This feels like duplicate code from the json model gen, we should make a helper for this